d2/src/main/java/com/linkedin/d2/discovery/stores/zk/ZooKeeperEphemeralStore.java

```diff
@@ -706,13 +747,24 @@ public void processResult(int rc, String path, Object ctx, List<String> children
       {
         case OK:
         {
+          if (_isInitialFetch) {
```

Comment:
Can there be race conditions if `emitSDStatusInitialRequestEvent` is taking a while, and there will be multiple initial request events sent? Shall we move the `_isInitialFetch = false` line before `emitSDStatusInitialRequestEvent`?

Reply:
it shouldn't currently since we have delayed resetting the next watch at line 722, but yeah, why not, will set it to false as the very first thing.